pkg/api/serve_test.go

```diff
@@ -121,7 +121,10 @@ func createDefaultAdminUser(t testing.TB, clt api.ClientWithResponsesInterface)
 func setupHandlerWithWalkerFactory(t testing.TB, factory catalog.WalkerFactory) (http.Handler, *dependencies) {
 	t.Helper()
 	ctx := context.Background()
-	viper.Set("blockstore.type", block.BlockstoreTypeMem)
+
+	if viper.Get(config.BlockstoreTypeKey) == nil {
+		viper.Set(config.BlockstoreTypeKey, block.BlockstoreTypeMem)
+	}
```

Comment:
The previous code assumed that the package level will use blockstore of mem type as global state - using `if` here insidate that we may change the state, which will make an issue when we will run some tests in parallel.
Will it be possble to setup the specific adater in a specific test without modify the viper global state?